src/hooks/useComposedCssClasses.tsx

```diff
@@ -22,7 +26,13 @@ export function useComposedCssClasses<
       return mergedCssClasses;
     }
     Object.keys(customClasses).forEach(key => {
-      mergedCssClasses[key] = twMerge(mergedCssClasses[key], customClasses[key]);
+      const builtIn = builtInClasses[key];
+      const custom = customClasses[key];
+      if (!builtIn || !custom) {
+        mergedCssClasses[key] = builtIn ?? custom;
+      } else {
+        mergedCssClasses[key] = twMerge(builtIn, custom);
```

Comment:
I haven't finish looking into the documentation for tailwind-merge at the time of the wip commit. Not sure if you saw this section about usage with custom tailwind config: https://github.com/dcastil/tailwind-merge#usage-with-custom-tailwind-config. Since we do have some custom styling defined in tailwind.config.cjs, I want to double check if that would may be handle differently.

Reply:
Ah yeah I missed that I'll check it out

Reply:
I added conflict resolution for the form-* classes. Custom colors and what not are automatically resolved by tailwind-merge, it's only new class groups like form-* we need to worry about. We don't need to worry about conflicts with a user's custom class groups, because we don't use any ones besides forms ourselves so there's no way to have a conflict.